src/mlpack/tests/recurrent_network_test.cpp

```diff
@@ -1413,4 +1413,68 @@ BOOST_AUTO_TEST_CASE(BRNNTrainReturnObjective)
   BOOST_REQUIRE_EQUAL(std::isfinite(objVal), true);
 }
 
+/**
+ * Test that RNN::Train() does not give an error for large rho.
+ */
+BOOST_AUTO_TEST_CASE(LargeRhoValueRnnTest)
+{
+  const int rho = 18;
+  const int hiddenSize = 128;
+  const int numLetters = 256;
+  using namespace arma;
+  using MatType = cube;
+  std::vector<std::string> trainingData;
+  trainingData.push_back(std::string("THIS IS THE INPUT"));
+
+  RNN<> model(rho);
+  model.Add<IdentityLayer<>>();
+  model.Add<LSTM<>>(numLetters, hiddenSize, rho);
+  model.Add<Dropout<>>(0.1);
+  model.Add<Linear<>>(hiddenSize, numLetters);
+
+  const auto makeInput = [](const char *line) -> MatType {
```

Comment:
To match the style guide, we should wrap the `{`.  (Not sure why the style checker didn't get that; maybe it doesn't read lambdas right.  Anyway I caught it so no problem :-D)